selenium_tests/base_page.py

```diff
@@ -0,0 +1,127 @@
+import logging
+from functools import partial
+
+from selenium.common.exceptions import (
+    NoSuchElementException,
+    InvalidSwitchToTargetException
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+class BasePageMeta(type):
+    """
+    This metaclass allows us to avoid writing boilerplate that just finds
+    an element in a page by a locator tuple.
+
+    Expected usage:
+
+        class SomePageLocators(object):
+            TITLE = (By.CSS_SELECTOR, '#title')
+            BUTTON = (By.CSS_SELECTOR, '#the_only_button')
+
+        class SomePage(BasePage):
+            locator_class = SomePageLocators
+            located_properties = ['title', 'button']
```

Comment:
I'd intended it as "the list of properties we create using locators", actually.  I think when we pull this into a separate project, I'm going to do away with this altogether (too easy to forget to add/remove things from it) in favor of using two Locator classes (one for properties, referenced from `locator_class`, another for the non-property locators).  For now, I'm inclined to leave this alone.